lib/core/typeinfo.nim

```diff
@@ -120,7 +133,7 @@ template `+!!`(a, b): untyped = cast[pointer](cast[ByteAddress](a) + b)
 proc getDiscriminant(aa: pointer, n: ptr TNimNode): int =
   assert(n.kind == nkCase)
   var d: int
-  var a = cast[ByteAddress](aa)
+  let a = cast[ByteAddress](aa)
   case n.typ.size
   of 1: d = ze(cast[ptr int8](a +% n.offset)[])
   of 2: d = ze(cast[ptr int16](a +% n.offset)[])
```

Comment:
```suggestion
  of 1: d = int(cast[ptr uint8](a +% n.offset)[])
  of 2: d = int(cast[ptr uint16](a +% n.offset)[])
```

I believe this is how you change `ze`. Also `d` in this proc could just be `result` I think.

Reply:
yes but let's defer this to followup PR